client/src/document.js

```diff
@@ -100,69 +100,93 @@ function RenderSideBar({ doc }) {
-        {content.map(node => {
-          if (node.content) {
-            return (
-              <li key={node.title}>
-                <SidebarLeaflets node={node} />
-              </li>
-            );
-          } else {
-            return (
-              <li key={node.uri}>
-                <Link to={node.uri}>{node.title}</Link>
-              </li>
-            );
-          }
-        })}
-      </ul>
-    </div>
-  );
-}
-
-function SidebarLeaflets({ node }) {
+function SidebarLeaf({ parent }) {
+  const titleNode = <h3>{parent.title}</h3>;
   return (
     <Location>
       {({ location }) => {
-        let hasActiveChild = false;
-        const listItems = node.content.map(childNode => {
-          const isActive = childNode.uri === location.pathname;
-          if (isActive && !hasActiveChild) {
-            hasActiveChild = true;
+        // walk through the tree, annotating its nodes:
+        // * `isActive` if a node contains the link to the current page
+        // * `open` if a node is on the path to the `isActive` node
+        function setOpenNodes(node, pathname) {
+          if (node.uri === pathname) {
+            node.open = true;
+            node.isActive = true;
+            return true;
+          }
+          if (node.content) {
+            for (const child of node.content) {
+              if (setOpenNodes(child, pathname)) {
+                node.open = true;
+                return true;
+              }
+            }
           }
-          return (
-            <li key={childNode.uri} className={isActive ? "active" : undefined}>
-              <Link to={childNode.uri}>{childNode.title}</Link>
-            </li>
-          );
-        });
+          return false;
+        }
+        setOpenNodes(parent, location.pathname);
 
         return (
-          <details open={!!hasActiveChild}>
-            <summary>{node.title}</summary>
-            <ol>{listItems}</ol>
-          </details>
+          <div>
+            {titleNode}
+            <ul>
+              {parent.content.map(node => {
+                if (node.content) {
+                  return (
+                    <li key={node.title}>
+                      <SidebarLeaflets node={node} />
+                    </li>
+                  );
+                } else {
+                  return (
+                    <li key={node.uri}>
+                      <Link to={node.uri}>{node.title}</Link>
+                    </li>
+                  );
+                }
+              })}
+            </ul>
+          </div>
         );
       }}
     </Location>
   );
 }
 
+function SidebarLeaflets({ node }) {
+  const listItems = node.content.map(childNode => {
+    if (childNode.content) {
+      return (
+        <li key={childNode.title}>
+          <SidebarLeaflets node={childNode} />
+        </li>
+      );
+    } else {
+      return (
+        <li
+          key={childNode.uri}
+          className={childNode.isActive ? "active" : undefined}
```

Comment:
```suggestion
          className={childNode.isActive && "active"}
```
that way you don't need to mention `undefined`. This works the same way and I think it looks cleaner

Reply:
Sorry if it seemed like I "attacked" your diff. Yes, I should have caught that in the original contribution when this was introduced. Sorry.